lib/Dialect/StaticLogic/StaticLogicOps.cpp

```diff
@@ -151,8 +151,11 @@ static LogicalResult verifyPipelineWhileOp(PipelineWhileOp op) {
                  "'staticlogic.pipeline.terminator' ops, found ")
              << inner;
 
-  // TODO: once there is an II attribute, verify that the II is satisfied by the
-  // definitions and uses of `iter_args`.
+  // Verify that the II is satisfied by the definitions and uses of `iter_args`.
+  if (op.II().hasValue() && op.II().getValue() < op.computeInitiationInterval())
```

Comment:
Could we include some additional information about what specific variable violated the requested initiation interval? (variable, use and def).